praw/models/comment_forest.py

```diff
@@ -72,12 +73,19 @@ def __len__(self) -> int:
         return len(self._comments)
 
     def _insert_comment(self, comment):
-        assert comment.name not in self._submission._comments_by_id
+        if comment.name not in self._submission._comments_by_id:
+            raise ClientException(
+                "You cannot replace MoreComments in a refreshed comment."
+            )
         comment.submission = self._submission
         if isinstance(comment, MoreComments) or comment.is_root:
             self._comments.append(comment)
         else:
-            assert comment.parent_id in self._submission._comments_by_id
+            if comment.parent_id in self._submission._comments_by_id:
```

Comment:
Let's leave this as an assertion as we have no way to reproduce it.

Reply:
But it's bad coding practice to leave assertions as logic checks, so maybe I'll have it raise just a plain Exception.

Reply:
I'll think you'll need a test case if you raise a proper exception.

Reply:
I'll just modify the assertion error.